Tests/AsyncQueueTests/FIFOQueueTests.swift

```diff
@@ -24,34 +24,29 @@ import XCTest
 
 @testable import AsyncQueue
 
-final class AsyncQueueTests: XCTestCase {
+final class FIFOQueueTests: XCTestCase {
 
     // MARK: XCTestCase
 
     override func setUp() async throws {
         try await super.setUp()
 
-        systemUnderTest = AsyncQueue()
-    }
-
-    override func tearDown() async throws {
-        try await super.tearDown()
-
-        await systemUnderTest.await { /* Drain the queue */ }
+        systemUnderTest = FIFOQueue()
     }
 
     // MARK: Behavior Tests
 
-    func test_async_sendsEventsInOrder() {
+    func test_async_sendsEventsInOrder() async {
         let counter = Counter()
         for iteration in 1...1_000 {
             systemUnderTest.async {
                 await counter.incrementAndExpectCount(equals: iteration)
             }
         }
+        await systemUnderTest.await { /* Drain the queue */ }
```

Comment:
We must drain the queue within the body of the test method in order to catch test failures. Test failures that occur after the test method has exited do not fail the test run.